hudi-common/src/main/java/org/apache/hudi/common/model/TaggableHoodieKey.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.model;
+
+import org.apache.hudi.common.util.Option;
+
+import javax.annotation.Nullable;
+
+/**
+ * Lean class to hold HoodieRecord information.
+ * This holds the HoodieKey along with the location information.
+ */
+public class TaggableHoodieKey extends HoodieKey {
+
```

Comment:
Isn't the combination of `HoodieKey` and `HoodieRecordLocation` an index item? The name `TaggableHoodieKey ` is confusing and why it inherits from a `HoodieKey`.

Reply:
don't quite get what you meant by "index item". The reasoning here is: we say "tag location" in the context of indexing so "taggable key" is a key that can be tagged to locations (current or new location) for the purpose of passing info to write status.

Reply:
For "index item" I'm refering to a record in `HoodieIndex`, the `HoodieIndex` is a mapping from recordKey to record location, which is more easy to understand, I don't think `TaggableXXX` is a good naming and I'm confused for the hierachy it has with `HoodieKey`. If it inherits from a `HoodieKey`, then it is a `HoodieKey`, which I think should be incorrect from the notion.

Reply:
i understood that you want composite pattern over inheritance - can you elaborate from the angle of disadvantages here? as in what is "incorrect from the notion".  For taggable as i explained is in the context of indexing, where we have "tagLocation" and "requiresTagging" as public APIs in `HoodieIndex`, do you still feel it's confusing?

Reply:
Now this is changed to `HoodieRecordDelegate` as a composite of objects